### ISSUE TITLE
Confusion about reference genome, strand and orientations.

### ISSUE THREAD

**Issue by carstenschwede** [lassefolkersen/impute-me#44]
First of all thanks a lot for providing an awesome tool!

While exploring some SNPs I came across differences between the 23andme and SNPedia data and apparently considering the reference genome in use is important when comparing data from different sources.

I wonder how this is handled in imputeMe:
If I understand correctly, after imputation and analysis, there are now three different datasets combined: Original source (e.g. 23andme), the imputation process and the SNPedia reference + further risk score data.

Do the differences in used reference genome affect the reliability of the end results? Are there any measures being taken in detecting the reference genome and/or strand used, e.g. when the data provider switched to a newer standard (I think this is the case with earlier 23andme versions)? Is there an automatic conversion taking place between them or do I need to take care of that by some form of manual conversion before (e.g. via DNA Tool Kit?)

Thanks!

**Comment by lassefolkersen** [comment]
Yes, excellent question. I love talking about strands and references. I think they are the source of ~80% of mistakes in DTC-genetics these days, because data formats are so diverse. I mean --- not just the columns with rs-ids and positions and genotypes, but more how e.g. indels are written, what strand they are. One vendor may have something as a C/T SNP while another has it as a G/A SNP, and _they are both correct_, if only not adhering to the same standard. But so many third party apps don't take this into account, and happily report that a user have several homozygote-alt BRCA1 variants or such.

At impute-me there's a strict adherence to the GRch37 1000 genomes v3 standard - meaning if an input-variant is found in data-set, with those alleles it is passed on to further processing - if not, a strand-flip is attempted - meaning e.g. C/T forced to G/A, and then passed on to further processing. The further processing (for microarrays, not for sequencings/vcf, that's another story) involves an imputation step. While this imputation step does do some work towards filling in blanks, it's de-facto main benefit actually is to do exactly what you ask about - it harmonise the input data to adhere to GRch37 1000 genomes v3 standard. So when things like that 23andme-version switch happens, it's actually not felt at all, because the above steps ensure that all analyzed data is anyway harmonized. 

So no it doesn't affect it, is the answer.

There's a very few, less than 0.1% of the samples case, typically combined DNA-kit submissions where the above harmonization steps become so confused that they result in errors. This is triggered at a set point for max strand-flip attempts, and the algorithms will complain that too many SNPs are not biallelic. But in those cases the whole sample fails. It's rare, but it does happen. And I've never seen it, except in combine-kit setups. So if you work with DNA kit studio and combine samples, this may happen - but you'll know because nothing comes out in the other end.

**Comment by carstenschwede** [comment]
Awesome. Thanks a lot for the quick and thorough answer!
